src/plugins/data_source/server/legacy/configure_legacy_client.ts

```diff
@@ -54,26 +64,47 @@ export const configureLegacyClient = async (
  * @returns child client.
  */
 const getQueryClient = async (
```

Comment:
seems a lot of duplicate code in `configure_client.ts`

Reply:
can you elaborate more? There's a `configure_client.ts` and `configure_legacy_client.ts`, are you referring both?

Reply:
yes, I mean the `configure_client.ts` and `configure_legacy_client.ts` has similar structure, and duplicate code, can we reuse code in these 2 files?

Reply:
I don't think there's a lot of duplicates, because interface is very different between new client and legacy client. I'd rather keep it, in case we want to deprecate legacy code in OSD in the future, MD could just delete the legacy folder, without impacting other code